### PR TITLE
Rework the creating tar vs. loading image logic

### PR DESCRIPTION
**Pull request by goldmann**

### CODE REVIEW

File: docker_scripts/squash.py [hunk 1]
```diff
@@ -399,6 +399,10 @@ def run(self):
             self.log.error("Image is not provided, exiting")
             sys.exit(1)
 
+        if not (self.output_path or self.load_image):
+            self.log.warn("No output path specified and loading into Docker is not selected either; squashed image would not accessible, proceeding with squashing doesn't make sense, exiting")
```
Comment (bkabrda) [inline docker_scripts/squash.py]: Did you mean to write "squashed image would not *be* accessible"?

Comment (goldmann) [inline docker_scripts/squash.py]: Of course this is what I wanted to say :)

**Comment by bkabrda** [comment]
Generally, I like this, there is however one aspect that I find problematic: I think you shouldn't call `sys.exit` when receiving wrong arguments, that seems to harsh for a library to do. I'd advise raising an exception (perhaps a `ValueError`) that can be caught by applications that use docker-scripts (rather then terminating such applications with `sys.exit`).

**Comment by goldmann** [comment]
That's a very good point and I'll rework it and fix the other `exit` in the same way.

**Comment by goldmann** [comment]
@bkabrda Can you take a look at this one more time?

File: docker_scripts/squash.py [hunk 2]
```diff
@@ -462,15 +459,13 @@ def run(self):
...
-            sys.exit(1)
+            raise SquashError("Preparing temporary directory failed")
 
         # Location of the tar with the old image
         old_image_tar = os.path.join(tmp_dir, "image.tar")
 
         # Save the image in tar format in the tepmorary directory
-        if not self._save_image(old_image_id, old_image_tar):
-            sys.exit(1)
+        self._save_image(old_image_id, old_image_tar)
```
Comment (bkabrda) [inline docker_scripts/squash.py]: Why wasn't the original `sys.exit` here replaced with `raise`?

**Comment by bkabrda** [comment]
I did one inline comment for a line where I wasn't sure whether you forgot to raise or you intentionally just removed the `sys.exit`. Otherwise this looks great to me.

File: docker_scripts/squash.py (hunk 2 above)
Comment (goldmann) [inline docker_scripts/squash.py]: The `_save_image`function raises `SquashError` on it's own.

Comment (bkabrda) [inline docker_scripts/squash.py]: Ah, I missed that. Cool, than I think this is ready to be merged!

**Comment by goldmann** [comment]
Done!

**Pull request merged**